reason-parser/src/reason_pprint_ast.ml

```diff
@@ -3375,8 +3399,8 @@ class printer  ()= object(self:'self)
     | PotentiallyLowPrecedence itm -> formatPrecedence ~loc:reducesAfterRight.pexp_loc itm
     | Simple itm -> itm
 
-  method ensureExpression expr ~reducesOnToken =
-    match self#unparseExprRecurse expr with
+  method ensureExpression ?(infixChain=false) ~reducesOnToken expr =
+    match self#unparseExprRecurse ~infixChain expr with
     | SpecificInfixPrecedence ({reducePrecedence; shiftPrecedence}, leftRecurse) ->
```

Comment:
This is a very good idea. An intermediate representation should solve the nuances of printing different infix applications. `~infixChain` felt very shaky from the start, I just couldn't think of something better at the moment 😄
I'll go ahead and implement it.